### ISSUE TITLE
Are TypeScript definitions generated by transpile supposed to include comments from the WIT file?

### ISSUE THREAD

**Issue by blaine-arcjet** [bytecodealliance/jco#397]
I found https://github.com/bytecodealliance/jco/pull/94 which fixed the TypeScript generation and it mentions comments, but I can't seem to get the comments I defined in my `.wit` file to actually show up when I use `jco transpile`. Is this something that is supported?

**Comment by guybedford** [comment]
Yes TypeScript output includes the comments from WIT. If you have an example of what specific doc comments are being missed in translation that would be great to track.

**Comment by blaine-arcjet** [comment]
Interesting! I'm not getting any comments but the current stuff isn't open source. Let me create a reproduction.

**Comment by sd2k** [comment]
I think I may just have run into this too, perhaps due to https://github.com/bytecodealliance/cargo-component/issues/229? Is there any way of pointing `transpile` to the original WIT file to get the comments back, since they're not included in the WASM right now?

**Comment by sd2k** [comment]
Huh, well I found a workaround: after running `jco transpile` I can run `jco types` and point it at the original WIT file to get the types with doc comments. For anyone finding this, if you have imports you'll need to add the corresponding WIT files to a `deps` directory next to the main WIT file, too.

**Comment by oovm** [comment]
- command

```sh
> jco -V
1.8.1
> jco transpile wit -o . --name inde
```

- Input: https://github.com/valkyrie-language/valkyrie.js/blob/master/projects/math-shim/wit/arithmetic.wit
- Output: https://github.com/valkyrie-language/valkyrie.js/blob/master/projects/math-shim/interfaces/wasi-math-arithmetic-f32.d.ts


Missing comment
